src/Microsoft.ComponentDetection.Detectors/pip/IPyPiClient.cs

```diff
@@ -84,54 +92,88 @@ public PyPiClient(IEnvironmentVariableService environmentVariableService, ILogge
+                    "Http GET at {ReleaseUrl} failed with status code {ResponseStatusCode}",
+                    release.Url,
+                    response.StatusCode);
+                return new List<PipDependencySpecification>();
+            }
 
-        var entry = package.GetEntry($"{name.Replace('-', '_')}-{version}.dist-info/METADATA");
+            // Store the .whl file on the disk temporarily
+            var tempFilePath = Path.GetTempFileName();
+            await using (var fileStreamWrite =
+                         new FileStream(tempFilePath, FileMode.Create, FileAccess.Write, FileShare.None))
+            {
+                await (await response.Content.ReadAsStreamAsync()).CopyToAsync(fileStreamWrite);
+            }
 
-        // If there is no metadata file, the package doesn't have any declared dependencies
-        if (entry == null)
-        {
-            return dependencies;
-        }
+            var content = new List<string>();
+            await using (var fileStreamRead =
+                         new FileStream(tempFilePath, FileMode.Open, FileAccess.Read, FileShare.None))
+            {
+                using var package = new ZipArchive(fileStreamRead);
 
-        var content = new List<string>();
-        using (var stream = entry.Open())
-        {
-            using var streamReader = new StreamReader(stream);
+                var entry = package.GetEntry($"{name.Replace('-', '_')}-{version}.dist-info/METADATA");
 
-            while (!streamReader.EndOfStream)
-            {
-                var line = await streamReader.ReadLineAsync();
+                // If there is no metadata file, the package doesn't have any declared dependencies
+                if (entry == null)
+                {
+                    return new List<PipDependencySpecification>();
+                }
+
+                await using var stream = entry.Open();
+                using var streamReader = new StreamReader(stream);
 
-                if (PipDependencySpecification.RequiresDistRegex.IsMatch(line))
+                while (!streamReader.EndOfStream)
                 {
-                    content.Add(line);
+                    var line = await streamReader.ReadLineAsync();
+                    if (line is null)
+                    {
+                        continue;
+                    }
+
+                    if (PipDependencySpecification.RequiresDistRegex.IsMatch(line))
+                    {
+                        content.Add(line);
+                    }
                 }
             }
-        }
 
-        // Pull the packages that aren't conditional based on "extras"
-        // Right now we just want to resolve the graph as most comsumers will
-        // experience it
-        foreach (var deps in content.Where(x => !x.Contains("extra ==")))
-        {
-            dependencies.Add(new PipDependencySpecification(deps, true));
-        }
+            File.Delete(tempFilePath);
```

Comment:
what if this fails?

The detector can run in parallel and potentially cause lockups.

Reply:
This code uses a different temporary path every time (`GetTempFileName`), so there won't be issues